subcommands/devices/cmd.go

```diff
@@ -59,11 +59,21 @@ func NewCommand() *cobra.Command {
 	updatesCmd.Flags().IntVarP(&listLimit, "limit", "n", 0, "Limit the number of updates displayed.")
 
 	cmd.AddCommand(configCmd)
-	cmd.AddCommand(updatesCmd)
+	addUuidCommand(cmd, updatesCmd)
+
 	return cmd
 }
 
-func getDeviceApi(_ *cobra.Command, name string) client.DeviceApi {
+func addUuidCommand(parent, command *cobra.Command) {
+	command.Flags().BoolP("by-uuid", "u", false, "Look up device by UUID rather than name")
+	parent.AddCommand(command)
```

Comment:
Just IMHO.

What I dislike here is that this function combines 2 concepts:
- Adding a flag.
- Adding a command to a parent.

Although it allows to cut 1 line of code in each subcommand, it looks like a violation of the _separation of concepts_.
Just IMHO the code looks less intuitive now, that is apparently my _subjective interpretation_.
But a stronger argument is that now applying 2 addXyzFlag functions (where needed) becomes more intricate, as one of them also does an extra task, and the other one must not do the same.

Reply:
there were several call sites where we were doing `cmd.AddCommand`, so the diff was growing in size to declare a local variable for the subcommand and then call AddCommand. I was also thinking it might help reduce the chance of someone forgetting to add the uuid-flag, but I don't think my approach lessens that.

I was about to the point of creating a zlast_alpabetical_file.go with an `init` function that is guaranteed to run last where I fix every subcommand. But that is just awful.

I'll clean this up to be more intentional.

Reply:
I just tried a slightly different approach at the end. I moved the commands over one by one to the getDevice function. Then in one small change I could add the `-by-uuid` flag to all commands from a single place. Its almost too clever - but I think it makes the chances of someone messing things up smaller.